fireapp/lib/presentation/volunteer_home/volunteer_home_view_model.dart

```diff
@@ -25,17 +27,28 @@ class VolunteerHomeViewModel extends FireAppViewModel {
 
   VolunteerHomeViewModel(
       this._authenticationRepository, this._shiftsRepository);
-
-  // write your functions and logic etc
-
-  //TODO retrieve shifts
   Future<void> fetchShifts() async {
+    _loadingState.add(RequestState.loading());
```

Comment:
You should be setting `_shifts` to `RequestState.loading()`, not this. Each RequestState stream is supposed to represent the lifecycle of a *single* operation (fetching some kind of data, posting a form, etc), so splitting it across multiple states isn't good practice. `_loadingState` isn't actually used here, so I'd appreciate it if you just deleted it completely.